src/main/java/oscar/essential/Parser.java

```diff
@@ -0,0 +1,73 @@
+package oscar.essential;
+
+import oscar.command.Command;
+import oscar.command.DeadlineCommand;
+import oscar.command.DeleteCommand;
+import oscar.command.EventCommand;
+import oscar.command.ExitCommand;
+import oscar.command.FindCommand;
+import oscar.command.ListCommand;
+import oscar.command.MarkCommand;
+import oscar.command.TodoCommand;
+import oscar.command.UnmarkCommand;
+import oscar.exception.OscarException;
+
+/**
+ * Handles raw user input and invokes the corresponding command.
+ */
+public class Parser {
+    /**
+     * Handles raw user input and invokes the corresponding command.
+     *
+     * @param fullCommand Raw user input.
+     * @return Corresponding command.
+     * @throws OscarException Invalid command.
+     */
+    public static Command parse(String fullCommand) throws OscarException {
+        String[] splits = fullCommand.split(" ", 2);
+        String command = splits[0].toLowerCase();
+        String details = fullCommand.length() > command.length() ? splits[1] : "";
+        switch (command) {
+        // Exit programme if user enters "bye" command
+        case "bye":
```

Comment:
Nice! Great to see that you have abided the indentation for switch statements, avoiding the common indentation mistake!